APP-63 Fix datacollection demo build

ohamov opened the pull request.

Commit:
APP-63 Fix datacollection demo build

ohamov commented:
/cc @forGGe @rasendubi

RostakaGmfun commented on datacollectiondemo/source/c/build.sh:
```diff
@@ -54,54 +51,23 @@ function select_arch {
+    cd $BUILD_DIR
+    cmake -DKAA_PLATFORM=$KAA_ARCH \
+          -DWITH_EXTENSION_EVENT=0 \
+          -DWITH_EXTENSION_CONFIGURATION=0 \
```
How about disabling notification extension as well?

RostakaGmfun commented on datacollectiondemo/source/c/build.sh:
```diff
@@ -54,54 +51,23 @@ function select_arch {
+          -DWITH_EXTENSION_EVENT=0 \
+          -DWITH_EXTENSION_CONFIGURATION=0 \
+          -DKAA_MAX_LOG_LEVEL=3 \
+          -DKAA_PLATFORM=$KAA_ARCH $KAA_TOOLCHAIN_PATH_SDK ..
```
KAA_PLATFORM is already specified a few lines above

Commit:
APP-63 Fix code review

forGGe commented:
+

Commit:
APP-63 Fix notification demo

rasendubi commented on datacollectiondemo/source/c/build.sh:
```diff
@@ -54,54 +51,24 @@ function select_arch {
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR"
+    cd $BUILD_DIR
+    cmake -DKAA_PLATFORM=$KAA_ARCH \
+          -DWITH_EXTENSION_EVENT=0 \
```
Use `OFF` here.

Dubland commented on datacollectiondemo/source/c/CMakeLists.txt:
```diff
@@ -15,23 +15,31 @@
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src
-                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/src
+                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/${KAA_SDK_DIR}/src
```
You shouldn't include SDK directory directly.

rasendubi commented on datacollectiondemo/source/c/CMakeLists.txt:
```diff
@@ -15,23 +15,31 @@
+set(KAA_SDK_DIR "client-c")
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR}
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
```
Why creating a separate directory?

ohamov replied:
Because It's convenient having tar.gz separately from unpacked sdk. It could be easily removed

rasendubi replied:
See comment below. It's better unpack to the build directory.

rasendubi commented on datacollectiondemo/source/c/CMakeLists.txt:
```diff
@@ -15,23 +15,31 @@
+               )
+execute_process(COMMAND sh -c "tar -zxf kaa-c*.tar.gz -C ${KAA_SDK_DIR}"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
+               )
```
Ideally, you should unpack SDK to the build directory, so yo don't modify source tree.

ohamov replied:
Agree with you. Also I think it is useful to keep build of sdk and sample app in separate directories.

rasendubi commented on datacollectiondemo/source/c/src/kaa_demo.c:
```diff
@@ -23,7 +23,7 @@
 #include <kaa/platform/kaa_client.h>
 #include <kaa/utilities/kaa_log.h>
-#include <kaa/kaa_logging.h>
+#include <extensions/logging/kaa_logging.h>
```
Wrong. Should be `<kaa_logging.h>`

ohamov replied:
Done

rasendubi commented on notificationdemo/source/c/CMakeLists.txt:
```diff
@@ -15,23 +15,31 @@
-
-project (C-SDK-sample)
-enable_language(C)
+project (C-SDK-sample LANGUAGES C)
```
`LANGUAGES` is not supported in cmake 2.8.8. Use `project(C-SDK-sample C)` instead.

ohamov replied:
Done

rasendubi commented on notificationdemo/source/c/src/kaa_demo.c:
```diff
@@ -23,7 +23,7 @@
 #include <kaa/platform/kaa_client.h>
 #include <kaa/utilities/kaa_log.h>
-#include <kaa/kaa_notification_manager.h>
+#include <extensions/notification/kaa_notification_manager.h>
```
Should be `<kaa_notification_manager.h>`.

ohamov replied:
Done

Commit:
APP-63 Improve build structure

The patch changes the build structure. The directories structure
of unpacked application looks like this one:

    - demo_application_directory
      - src
      -libs
        -kaa
          - kaa-c*.tar.gz
      CMakeList.txt
      build.sh
      README.txt

After the build procedure been performed the build directory is
created on the top level has the following structure:
    - build
      - client-c
         - 'sdk source files'
         - build
           - 'built sdk components'
           - 'sdk build artifacts'
         - demo client
         - 'demo build artifacts'

ohamov commented:
@rasendubi take a look please

rasendubi commented:
LGTM

forGGe commented:
+
